onadata/libs/utils/export_tools.py

```diff
@@ -163,9 +163,17 @@ def generate_export(export_type, xform, export_id=None, options=None,
     if options.get("dataview_pk"):
         dataview = DataView.objects.get(pk=options.get("dataview_pk"))
         records = dataview.query_data(dataview, all_data=True)
+        total_records = dataview.query_data(dataview,
+                                            count=True)[0].get('count')
     else:
         records = query_data(xform, query=filter_query, start=start, end=end)
 
+        if filter_query:
+            total_records = query_data(xform, query=filter_query, start=start,
+                                       end=end, count=True)[0].get('count')
```

Comment:
well, since count is `True`, looks like this will be a list of length 1 unless the query results in a generator, in which case maybe you'd need to call `len` on it? Is this guaranteed to be on line 15 of the `apps.viewer.models.parsed_instance` function? Not related, but why does the condition on lines 10-11 of `apps.viewer.models.parsed_instance` exist at all? this would put a lot of objects in memory and probably segfault on large datasets